distribution/lib/Standard/Database/0.0.0-dev/src/Internal/DB_Data_Link_Helpers.enso

```diff
@@ -12,3 +17,56 @@ data_link_connection_parameters (source : Data_Link_Source_Metadata) -> Connecti
     audit_mode = if Enso_User.is_logged_in then "cloud" else "local"
     options_vector = [["enso.internal.audit", audit_mode]] + (if related_asset_id.is_nothing then [] else [["enso.internal.relatedAssetId", related_asset_id]])
     Connection_Options.Value options_vector
+
+## PRIVATE
+save_table_as_data_link table destination on_existing_file:Existing_File_Behavior =
+    data_link_setup = table.connection.data_link_setup.if_nothing <|
+        Error.throw (Illegal_Argument.Error "Saving a Table as data link is currently not supported in this backend.")
+
+    # For a trivial query we return the table name.
+    link_type = if table.is_trivial_query then ..Table table.name else
+        sql_statement = table.to_sql
+        prepared = sql_statement.prepare
+        # If there are no interpolations, we can do a raw query represented by Text (more friendly in the GUI).
+        if prepared.second.is_empty then ..Query prepared.first else
+            # Lastly, if there are interpolations, we need to fully serialize
+            ..SQL_Statement sql_statement
+    result = data_link_setup.save_as_data_link destination on_existing_file link_type
+    referred_temporary_tables = _find_referred_temporary_tables table.connection table.context
+    if referred_temporary_tables.is_nothing then result else
+        warning = Illegal_State.Error "The saved query seems to refer to tables "+referred_temporary_tables.to_text+" which are temporary. Such tables may cease to exist once the session is closed, so the saved data link will no longer be valid and will fail to open."
+        Warning.attach warning result
```

Comment:
Ah I wanted to mention it, but I forgot.

In the doc comment I noted:
```
   Note that this is a heuristic and it may potentially lead to false positives
   if aliasing table names exist across schemas. Supporting tables with clashing
   names across schemas is something that may need to be revised overall in the
   Database library.
```

Indeed I'm not 100% sure of our table mapping in cases where one has two tables with the same name across two accessible schemas. This heuristic is simple and it could lead to false positives. If we do more 'reading tables across schemas' it'd probably have to be updated.

That's why I wanted to make it only a warning. If the warning is a false positive, it can be dismissed and one can continue working.

But if it were an error, it could prevent users from creating a valid datalink.

Reply:
Of course, this is not ideal - ideally we'd fix the heuristic to make sure it always works - and then error would make sense.

But I'm not sure if I know how to fix this heuristic properly. The whole idea of warning on temporary tables is something I added as an addition to this PR. Trying to fix the heuristic to be 100% accurate is well out of scope of this PR as it would require much more work and will probably be backend-specific. This was making me reconsider if I should add this check at all, but I decided that a relatively accurate check as a warning is better than no check at all.

Reply:
Alternatively, I can revert the check for now and create a ticket to create a more accurate implementation - as it can take more time to do it, so it would need to be scheduled.